app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/BaseFragment.kt

```diff
@@ -19,16 +17,36 @@ abstract class BaseFragment<BINDING: ViewBinding>(
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        _binding = inflaterFunction.invoke(inflater, container, false)
+        _binding = onCreateViewBinding(inflater, container, savedInstanceState)
         return binding.root
     }
 
+    abstract fun onCreateViewBinding(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): BINDING
+
+    fun setToolbarTitle(title: String = "") {
+        (activity as MainActivity).setToolbarTitle(title)
+    }
+
+    fun showToolbarUpButton() {
+        (activity as MainActivity).setToolbarUpButtonVisible(true)
```

Comment:
I think inside the MainActivity you can create some addOnDestinationChangeListener and make it visible/invisible from there based on fragment . Don't forget to make the logic for it inside the MainActivity viewModel ( which you will have to create as it doesn't exist now ) 

`val navHostFragment = supportFragmentManager
            .findFragmentById(R.id.nav_host) as NavHostFragment
        val navController = navHostFragment.navController
        navController.addOnDestinationChangedListener { controller, destination, arguments -> }
`